### ISSUE TITLE
Fractional values in gpu (resource_per_trial)

### ISSUE THREAD

**Issue by alokgogate** [ray-project/tune-sklearn#143]
Is it possible to give fractional values along with parameter `use_gpu` while using `TuneSearchCV` ?

[Labels]: question

**Comment by richardliaw** [comment]
Hmm, currently not possible to do so, but let me tag this as a feature request (and it should be easy to implement!)

**Comment by Yard1** [comment]
I can work on this.

**Comment by richardliaw** [comment]
That's awesome - it should be really straightforward (see n_cpus)

**Comment by Yard1** [comment]
Should we add a new parameter for that, or allow users to pass integers to `use_gpu`, in which case the same logic as for `n_jobs` will be used (see line 531 in `tune_basesearch.py`)?

**Comment by richardliaw** [comment]
Hmm I think you just want to make n_jobs also configure fractional gpu (no
new param)

**Comment by richardliaw** [comment]
Hey @alokgogate, @Yard1 just merged this -- could you give it a try?

```
TuneSearchCV(use_gpu=True, n_jobs=8)
```

will use 8 jobs with the proper CPU/GPU fractions :)

**Comment by richardliaw** [comment]
I'll close this for now, but definitely let me know if this works for you (and feel free to reopen/open a new issue!)

**Comment by alokgogate** [comment]
Hi @richardliaw, 

I tried to install it using the master branch 
the setup.py get's stuck and terminates after some time with the following error 👎 
```
Couldn't find index page for 'parameterized' (maybe misspelled?)
Scanning index of all packages (this may take a while)
Reading https://pypi.python.org/simple/
No local packages or download links found for parameterized
error: Could not find suitable distribution for Requirement.parse('parameterized')
```

I check for package `parameterized` and have it already installed

**Comment by richardliaw** [comment]
Hmm maybe theres a pip outage - can you try commenting it out on setup.py?

**Comment by alokgogate** [comment]
didn't find pip outage or outage in the file

**Comment by richardliaw** [comment]
sorry, can you just try commenting out "parametrize" in the setup.py?

**Comment by alokgogate** [comment]
Don't see that one either

**Comment by richardliaw** [comment]
hmm @alokgogate I just merged a fix, can you try: `pip install git+https://github.com/ray-project/tune-sklearn`

**Comment by alokgogate** [comment]
Hi @richardliaw, this worked and was able to install and run a few experiments. I have a few observations though:

When I run with `n_jobs=8,use_gpu=True` and set `verbose=2` I can see that a fraction of GPU is requested (I have 4 GPU cards hence each experiment is requesting 0.5). But i dont see the GPU's being used when look at nvidia-smi 
Is this expected behaviour? I am using xgboost to test this and have set `tree_method="gpu_hist"` 
Would this parameter become redundant given that `use_gpu=True` is set?

**Comment by richardliaw** [comment]
Hmm, you need to set `gpu_hist` for xgboost.
